pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java

```diff
@@ -109,8 +109,30 @@
          * Keep in mind that this NOT a hard threshold, as other tables can also be assigned to this server, and that in
          * certain conditions (eg. if the number of servers, replicas of Kafka partitions changes) where Kafka partition
          * to server assignment changes, it's possible to end up with more (or less) than this number of rows in memory.
+         *
+         * If this value is set to 0, then the consumers adjust the number of rows consumed by a partition such that
+         * the size of the completed segment is the desired size (see REALTIME_DESIRED_SEGMENT_SIZE), unless
+         * REALTIME_SEGMENT_FLUSH_TIME is reached first)
          */
         public static final String REALTIME_SEGMENT_FLUSH_SIZE = "realtime.segment.flush.threshold.size";
+        /*
+         * The desired size of a completed realtime segment.
+         * This config is used only if REALTIME_SEGMENT_FLUSH_SIZE (or REALTIME_SEGMENT_FLUSH_SIZE  + ".llc") is set
+         * to 0. Default value of REALTIME_SEGMENT_FLUSH_SIZE is "200M". Values are parsed using DataSize class.
+         *
+         * The value for this configuration should be chosen based on the amount of memory available on consuming
+         * machines, the number of completed segments that are expected to be resident on the machine and the amount
+         * of memory used by consuming machines. In other words:
+         *
+         *    numPartitionsInMachine * (consumingPartitionMemory + numPartitionsRetained * REALTIME_SEGMENT_FLUSH_SIZE)
```

Comment:
should this be REALTIME_DESIRED_SEGMENT_SIZE in the formula in the comment?
numPartitionsInMachine * (consumingPartitionMemory + numPartitionsRetained * REALTIME_DESIRED_SEGMENT_SIZE)

Reply:
yes, will fix.